File: src/ClientApp/src/app/dtos/profile/CreateProfileDto.ts
import {Gender} from "../../constants/gender";

export interface CreateProfileDto {
  name: string;
  lastName?: string;
  birthDate: Date;
  gender: Gender;
  bio?: string;
  height?: number;
  showAge: boolean;
  ageFrom: number;
  ageTo: number;
  maxDistance: number;
  preferredGender: Gender;
  goalId?: number;
  cityId: number;
  userId: string;
}

File: src/matchMaker.Services/Match/Match.BusinessLogic/DTOs/Profile/CreateProfileDto.cs
using Common.Constants;

namespace Match.BusinessLogic.DTOs.Profile;

public class CreateProfileDto
{
    public long Id { get; set; }
    public string Name { get; set; }
    public string LastName { get; set; }
    public DateTime BirthDate { get; set; }
    public int AgeFrom { get; set; }
    public int AgeTo { get; set; } 
    public Gender Gender { get; set; }
    public Gender PreferredGender { get; set; }
    public int MaxDistance { get; set; }
    public string Country { get; set; }
    public string City { get; set; }
    public string MainImageUrl { get; set; }
}